src/components/notifications/Message.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * Displays a notification message with support for navigating to another page,
+ * or some other action, when certain message types are clicked / tapped.
+ *
+ * @author psarando
+ */
+import React from "react";
+
+import Link from "next/link";
+
+import { getDisplayMessage } from "./utils";
+
+import NotificationCategory from "components/models/NotificationCategory";
+import SystemId from "components/models/systemId";
+
+import { getAnalysisDetailsLinkRefs } from "components/analyses/utils";
+import { getListingPath } from "components/apps/utils";
+import { getFolderPage, getParentPath } from "components/data/utils";
+import NameLink from "components/utils/NameLink";
+
+import { Typography } from "@material-ui/core";
+
+function MessageLink(props) {
+    const { message, href, as } = props;
+
+    return (
+        <Link href={href} as={as} passHref>
+            <NameLink name={message} />
+        </Link>
+    );
+}
+
+function AnalysisLink(props) {
+    const { notification } = props;
+
+    const message = getDisplayMessage(notification);
+    const action = notification.payload?.action;
+
+    const isJobStatusChange = action === "job_status_change";
+    const isShare = action === "share";
+    if (isShare || isJobStatusChange) {
+        const analysisId =
+            isShare && notification.payload?.analyses?.length > 0
+                ? notification.payload.analyses[0].analysis_id
+                : isJobStatusChange && notification.payload?.id;
+
+        if (analysisId) {
+            const [href, as] = getAnalysisDetailsLinkRefs(analysisId);
+
+            return <MessageLink message={message} href={href} as={as} />;
+        }
+    }
+
+    return message;
+}
+
+function AppLink(props) {
+    const { notification } = props;
+
+    const message = getDisplayMessage(notification);
+    const action = notification.payload?.action;
+
+    if (action === "share" && notification.payload?.apps?.length > 0) {
+        const app = notification.payload.apps[0];
+
+        const href = getListingPath(
+            null,
+            null,
+            null,
+            null,
+            null,
+            JSON.stringify({
+                system_id: app.system_id || SystemId.de,
+                id: app.category_id,
+            })
+        );
+
+        return <MessageLink message={message} href={href} />;
+    }
+
+    return message;
+}
+
```

Comment:
When I tried this locally, clicking on an `App` Sharing notification caused an `400` error. The address in URL bar was 
`http://localhost:3000/apps?selectedOrder=null&selectedOrderBy=null&selectedPage=null&selectedRowsPerPage=null&selectedFilter=null&selectedCategory={%22system_id%22:%22de%22,%22id%22:%22eeeeeeee-eeee-eeee-eeee-eeeeeeeeeeee%22}`
![image](https://user-images.githubusercontent.com/1250790/97460666-58cf8480-1913-11eb-9736-c0fe1657a71d.png)



I think Clicking on an app notification should take users to the individual app listing using this following `href` and `as`

const href = `/${NavigationConstants.APPS}/[systemId]/[appId]`;	
    const as = `/${NavigationConstants.APPS}/${selectedApp?.system_id}/${selectedApp?.id}`;

Reply:
So I was attempting to use `components/apps/utils/getListingPath` by passing `null` for all params except the `category` param, hoping that the code that built or parsed the URL would strip params with `null` values, but it looks like this function is currently building a raw string for the URL and these nulls become the string "null" when parsed by the target page, and I didn't test this in dev mode before opening this PR 😞 

I agree that this link should take the user to the specific app in the listing, so I've made the suggested change, but first I was curious and tried refactoring `getListingPath` to return a URL `{pathname, query}` object compatible with `next/link` and `Router.push`, but it appears `next/link` also preserves `null` values set in that URL's `query` object 😞